packages/babel-parser/src/tokenizer/index.ts

```diff
@@ -82,6 +82,8 @@ export class Token {
   declare loc: SourceLocation;
 }
 
+let locDataCache: Uint32Array;
```

Comment:
Is the location data cache shared between different parser inputs?

Reply:
Yes, actually this is just sharing a `Uint32Array` instance to avoid the time-consuming creation of a new one, the data inside is not shared.

Reply:
IIRC the data inside is correct if and only if for every `getLoc` call, there is a `setLoc` call preparing the location data for the exact `index`. If the `getLoc` call is issued before `setLoc`, it will read 1) uninitialized locDataCache cells or 2) location data used by previous inputs. This is prone to errors because of the implicit constraint here that if one should throw an error at the location, the location must have been prepared by `setLoc`, and there will be no immediate errors if it happens that the previous input gives the same location data.

I would suggest we zero-filling the `locDataCache` in the `parse()` entry and check if they are zero when `getLoc` is called, if they are zero, we should throw an internal error reminding contributors to manually add the `setLoc` call.

Reply:
I added it in the non-release version.
Because it affects performance and the user may still be able to use it normally even if the location information is wrong.